api/api.go

```diff
@@ -660,8 +664,19 @@ func NewClient(config *Config) (*Client, error) {
 		}
 	}
 
+	var addr, pref string
+
 	parts := strings.SplitN(config.Address, "://", 2)
 	if len(parts) == 2 {
+		switch parts[0] {
+		case "http", "https":
+			// for http and https addresses check if address contains prefix path.
+			parts := strings.SplitN(parts[1], "/", 2)
```

Comment:
I'm curious if we should be using `url.Parse` here instead? I realize we use `strings.SplitN` elsewhere in this package, but in this case, since we need to deal with both the address and the protocol prefix, I'm wondering if we should just use functionality from the standard library?

Reply:
Thanks for the response. It's a good point, haven't thought about that. I think using `url.Parse` makes code a bit clearer and it shouldn't increase the PR size overall.

Reply:
Were you still planning to make this change?

Reply:
@eculver working on it.

Reply:
Tried, the issue is `url.Parse` doesn't allow to parse schemeless address, which `parts[1]` is. It's possible to add schema here, but it's pointless, it's more clear and easier to leave this code "as is".

Reply:
Got it. In this case, can we add a comment about the situation and then, if possible, just adding a test and we can get this merged.